src/components/TradeWidget/index.tsx

```diff
@@ -690,8 +717,46 @@ const TradeWidget: React.FC = () => {
     receiveValue,
   ])
 
+  const fallBackNetworkId = networkId || Network.Mainnet // fallback to mainnet
+
+  const { needToAddSellToken, needToAddReceiveToken } = useMemo(() => {
+    const needToAddSellToken =
```

Comment:
A bit repetitive. Is it possible to return the array of addresses to add, so we also remove the repetition in the JSX part?

Reply:
Please elaborate

Reply:
instead of returning:
```
{
      needToAddSellToken,
      needToAddReceiveToken,
    }
```

you return an array of addresses. So this array would be:
* empty
* one address
* two addresses

Then I say is repetitive, cause line 723 and 728 expressions looks very ilike. 
Then in https://github.com/gnosis/dex-react/blob/43b68b7f503ad7a413512c45cd54dcc0951e9d94/src/components/TradeWidget/index.tsx#L746 , lines 746 and 753 are also repetitive, and if you have the array I mentioned above, you wouldn't have that repetition.